src/service.rs

```diff
@@ -1507,13 +1508,14 @@ impl Service {
             match active_request.request_body {
                 // if a failed FindNodes request, ensure we haven't partially received packets. If
                 // so, process the partially found nodes
-                RequestBody::FindNode { .. } => {
-                    if let Some(nodes_response) = self.active_nodes_responses.remove(&node_id) {
+                RequestBody::FindNode { ref distances } => {
+                    if let Some(nodes_response) = self.active_nodes_responses.remove(&id) {
                         if !nodes_response.received_nodes.is_empty() {
-                            warn!(
-                                "NODES Response failed, but was partially processed from: {}",
-                                active_request.contact
-                            );
+                            let node_id = active_request.contact.node_id();
+                            let addr = active_request.contact.socket_addr();
+                            let received = nodes_response.received_nodes.len();
+                            let expected = distances.len();
+                            warn!(%node_id, %addr, %error, %received, %expected, "FINDNODE request failed with partial results");
```

Comment:
isn't this wording a bit misleading? distances and nodes don't map 1-1. we can receive more than one node in the response at any distance in the requested range of distances.

```suggestion
                            let received = nodes_response.received_nodes.len();
                            warn!(%node_id, %addr, %error, %received, ?distances, "FINDNODE request failed with partial results");
```